### DISCUSSION TITLE
Joomla 4 - ordering of Javascripts

### DISCUSSION THREAD

**Discussion by PhocaCz** [joomla/joomla-cms#38248] [Q&A] [answered]
Hi,

adding following code:

```
$wa = $app->getDocument()->getWebAssetManager();
$wa->registerAndUseScript('com_extension.something', 'media/com_extension/js/xxx.min.js', [], [], ['core']);
$wa->registerAndUseScript('com_extension.somethinginitialize', 'media/com_extension/js/xxx.initialize.js', [], [], ['core', 'com_extension.something']);
```

loads the Javascripts with the following order:

xxx.initialize.js
xxx.min.js

Is there a way to set ordering so the order is following:

xxx.min.js
xxx.initialize.js

The only way I can achieve this is to rename xxx.min.js to xxx.ain.js (alphabetically is a before i), but that would make it look like it's alphabetical ordering, which would be very weird.

Thank you, Jan

**Comment by Fedik** [comment, marked as answer]
You mean that both scripts have the same `xxx` base name?
Can you please do next: enable debug and check source code, what asset name there will be for  each of this script? (look for `data-asset-name` in script tag) 

It already should work like that. If not then I suspect some strange bug in `HTMLHelper::script`.

**Comment by PhocaCz** [comment]
Hi, thank you for the answer, testing with enabled debug mode and everything is OK. data-asset-name and data-asset-dependencies are correct, even the ordering is correct. I did a fresh install of Joomla and even there everything is OK. So it seems like I need to check system plugins, etc., if there is something which changes the output.

As it works for me on clean install, I think, it is OK in core, so this topic can be closed. I will investigate more on my current instance.

Thank you very much.

Jan
